### ISSUE TITLE
Support internal registry from .npmrc file

### ISSUE THREAD

**Issue by thecodejack** [vscode-contrib/vscode-versionlens#38]
I have a custom registry set in `.npmrc` file. So it is throwing 404 error for internal packages. Following is how my `.npmrc` file looks like.

```
registry=http://internal.nexus.com/nexus/content/groups/npm-all/
@types:registry=https://registry.npmjs.com
```

**Comment by pflannery** [comment]
versionlens uses the `npm view` command to acquire package versions so when running `$npm view bootstrap` in a directory that contains an `.npmrc` file (configured like the above) I also see a 404 error.

Might be able to get around this by adding a new app config option which would let a user specify a different directory to run `npm view` from.

**Comment by thecodejack** [comment]
yes. When I run `npm view` from my project, I am not getting the 404 error. Not sure about the dynamics of vscode, I would suggest to run `npm view` from project path if there is a possibility.

**Comment by thecodejack** [comment]
or we can even run `npm view` where package.json is found.

**Comment by thecodejack** [comment]
Thanks. I would also suggest to take the path where package.json file present rather workspace path. Coz there is a possibility that many workspace different to package.json file.

**Comment by pflannery** [comment]
@thecodejack damn I must of switched off last night when I did this. 
I've now noticed that this change was causing npm to create an `etc` folder (like this issue refers to https://github.com/npm/npm/issues/11802)
I've my rolled my changes back to previous commit

I can't find another way to change the cwd when using the npm progmatic api so might have to dump it and run npm as child process so that a cwd can be specified (this might make it much slower)

**Comment by joelbourbon** [comment]
Hey guys,

Just posting here to revive this issue. I just got your extension and experienced this limitation. Would be awesome to get this fixed.

I Might take a look and try to fix this issue. Never worked with VS Code extensions but it's worth a try 😄 

Thanks,
CC @cstlaurent

**Comment by thecodejack** [comment]
Have resolved this temporarily by setting the registry url globally when needed.

**Comment by josundt** [comment]
Please fix!

**Comment by acidgh** [comment]
We're also using a custom, private registry in my company, and as soon as a private dependency of our own is listed in the *package.json*, the plugin fails to retrieve any version.

Do you think plugin will ever support custom registries? Thanks 👍

**Comment by pflannery** [comment]
The bug I orginally had when setting the `--prefix` flag has been fixed (Bug info [here](https://github.com/npm/npm/pull/20212#issuecomment-378177943)) 

I have now re-implemeted a fix to support .npmrc files. The fix sets the --prefix flag to the directory of the currently open package.json directory. 

Please could someone install [0.22.0-beta.1](https://github.com/vscode-contrib/vscode-versionlens/releases/tag/0.22.0-beta.1) and confirm that .npmrc files are now fixed for them?

**Comment by pflannery** [comment]
This should be fixed in [v0.22.0](https://github.com/vscode-contrib/vscode-versionlens/blob/master/CHANGELOG.md#0220)

**Comment by acidgh** [comment]
Seems indeed to be working correctly on 0.22.0, thanks a lot
